Frameworks/Foundation/NSOperationQueue.mm

```diff
@@ -30,167 +30,320 @@
 #import "LoggingNative.h"
 #import "NSOperationInternal.h"
 
+#include <deque>
+#include <mutex>
+#include <condition_variable>
+#include <algorithm>
+#include <memory>
+
 static const wchar_t* TAG = L"NSOperationQueue";
 
-typedef void* gpointer;
+class NSOperationQueuePriv {
+public:
+    NSOperationQueuePriv(NSOperationQueue* parent) : _thread(nullptr), _maxConcurrentOperationCount(1), _isSuspended(NO) {
```

Comment:
Note for eventual future implementer: The C++ class is providing no benefit. Objective-C++ classes can contain C++ things, and these aren't being custom-initialized.